packages/react-components/react-tree/src/hooks/useFlatTree.ts

```diff
@@ -0,0 +1,307 @@
+  /**
+   * efficiently gets a single item by id instead of navigating through a list.
+   * if no id is provided the virtual root item is returned instead
+   */
+  getItem(id?: string): FlatTreeItem | null;
+  /**
+   * all items available without properly filtering by visibility,
+   * use `.map` or `.toArray` instead to get visible items
+   * @internal
+   */
+  dangerouslyUnfilteredItems: FlatTreeItem[];
+  /**
+   * internal method used to react to an `onNavigation` event.
+   * This method ensures proper navigation on keyboard and mouse interaction.
+   * In case of virtualization it might be required to cancel default provided `onNavigation`
+   * event and then call this method manually.
+   *
+   * @example
+   * ```ts
+   * // react-window
+   * const handleNavigation = (event, data) => {
+   *   event.preventDefault();
+   *   const nextItem = tree.getNextNavigableItem(data);
+   *   // scroll to item using virtualization scroll mechanism
+   *   if (nextItem && !targetDocument.getElementById(nextItem.id)) {
+   *     listRef.current.scrollToItem(nextItem.index);
+   *   }
+   *   // wait for scrolling to happen and then invoke navigate method
+   *   requestAnimationFrame(() => {
+   *     tree.navigate(data);
+   *   });
+   * };
+   *```
+   */
+  navigate(data: TreeNavigationData_unstable): void;
+  /**
+   * returns next item to be focused on a navigation.
+   * This method is provided to decouple the element that needs to be focused from
+   *  the action of focusing it itself.
+   */
+  getNextNavigableItem(data: TreeNavigationData_unstable): FlatTreeItem | null;
+  /**
+   * returns all visible items as an array.
+   *
+   * almost equivalent to:
+   * ```js
+   * tree.map(item => item)
+   * ```
+   */
+  toArray(): FlatTreeItem[];
+  /**
+   * behaves similar to [Array.map](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/map).
+   * This method is provided for performance purposes,
+   * it reduces by one the amount of iterations required to map from a tree item to the next mapped value.
+   *
+   * @example
+   *```js
+   * // 2 iterations, 1 to create initial array and another 1 to map
+   * tree.toArray().map(item => item.id)
+   * // 1 iteration, it maps  at the same time it's creating
+   * tree.map(item => item.id)
+   * ```
+   */
+  map<NextValue>(fn: (child: FlatTreeItem) => NextValue): NextValue[];
+};
+
+/**
+ * this hook provides FlatTree API to manage all required mechanisms to convert a list of items into renderable TreeItems
+ * in multiple scenarios including virtualization.
+ *
+ * !!A flat tree is an unofficial spec for tree!!
+ *
+ * It should be used on cases where more complex interactions with a Tree is required.
+ * On simple scenarios it is advised to simply use a nested structure instead.
+ *
+ * @param items - a list of tree items
+ * @param options - in case control over the internal openItems is required
+ */
+export function useFlatTree_unstable(
```

Comment:
has any of the logic changed? or is this just a copy+paste from  `useFlatTree`?

Reply:
Yes, the logic has changed a bit, a more comprehensive API has been exported from this hook

Reply:
I'll remove this from this PR, since it's not pertinent to roving tabindex.